static/js/containers/CollectionDetailPage.js

```diff
@@ -215,12 +238,25 @@ const mapStateToProps = (state, ownProps) => {
   }
 }
 
-export default R.compose(
+const ConnectedCollectionDetailPage = R.compose(
   connect(mapStateToProps),
-  withDialogs([
-    { name: DIALOGS.COLLECTION_FORM, component: CollectionFormDialog },
-    { name: DIALOGS.EDIT_VIDEO, component: EditVideoFormDialog },
-    { name: DIALOGS.SHARE_VIDEO, component: ShareVideoDialog },
-    { name: DIALOGS.DELETE_VIDEO, component: DeleteVideoDialog }
```

Comment:
@mbertrand 

The reason for adding (1) `getDialogComponent`, and (2) `ConnectedCollectionDetailPage` is to simplify testing.

Initially I did use your suggestion:
```
{ name: DIALOGS.ANALYTICS, component: AnalyticsDialog }
```

But with that approach, it was difficult to test the `CollectionDetailPage` component in `CollectionDetailPage_test.js`.

The difficulties arose from having to mock methods/api in `CollectionDetailPage_test.js` related related to analytics components.

I thought it would be bad to mock the analytics methods in `CollectionDetailPage_test.js`, so I added the `getDialogComponent`, and unconnected `CollectionDetailPage` component, to enable mocking of dialogs in `CollectionDetailPage_test.js`.

For example:
https://github.com/mitodl/odl-video-service/pull/519/files#diff-1abcece8e6e13d39a500dd59c2fc93eaR197

With this in mind, do you think we should:
A. keep the `ConnectedCollectionDetailPage`/`getDialogComponent` approach
B. revert to the `{ name: DIALOGS.ANALYTICS, component: AnalyticsDialog }` and mock the analytics logic in `CollectionDetailPage_test.js`
C. something else?

Reply:
In general I'd prefer to avoid adding complexity to application code for unit testing purposes, but I'm fine with keeping option A, assuming that option B would require the unit tests to be rewritten with a different approach.

Reply:
Roger that, I'll go w/ option A & leave it as-is.